third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -846,4 +865,12 @@ impl<'env> ModuleContext<'env> {
         }
         result
     }
+
+    /// Converts to a name with location as expected by the SourceMap format.
+    pub(crate) fn source_name(&self, name: impl AsRef<Symbol>, loc: impl AsRef<Loc>) -> SourceName {
```

Comment:
Are we expecting some other things that will be convertable to `&Symbol` that might show up here?  I realize `AsRef<Loc>` was already fairly common in the code, and `AsRef<str>` is very useful in Rust, but do we anticipate other implementations that would warrant slightly more complex code here?

Reply:
Its really just about references, and we should have used it more frequently. With this you are able to pass parameters without `*` or `&`. We often need to do `*` because in matches, we get `&Symbol`. Similar for locations, we often need to add `&`. This reduces the noise a bit.